bundles/automation/org.eclipse.smarthome.automation.module.script.rulesupport/src/main/java/org/eclipse/smarthome/automation/module/script/rulesupport/internal/RuleSupportScriptExtension.java

```diff
@@ -0,0 +1,198 @@
+import org.slf4j.LoggerFactory;
+
+/**
+ * This Script-Extension provides types and presets to support writing Rules using a ScriptEngine.
+ * One can write and register Rules and Modules by adding them through the HandlerRegistry and/or RuleRegistry
+ *
+ * @author Simon Merschjohann
+ *
+ */
+public class RuleSupportScriptExtension implements ScriptExtensionProvider {
+    private final static Logger logger = LoggerFactory.getLogger(RuleSupportScriptExtension.class);
+
+    private RuleRegistry ruleRegistry;
+
+    private static HashMap<String, Collection<String>> presets = new HashMap<>();
+
+    private static HashMap<String, Object> staticTypes = new HashMap<>();
+    private static HashSet<String> types = new HashSet<String>();
+
+    private ConcurrentHashMap<String, HashMap<String, Object>> objectCache = new ConcurrentHashMap<>();
+
+    private ScriptedRuleProvider ruleProvider;
+
+    private ItemRegistry itemRegistry;
+
+    static {
+        staticTypes.put("SimpleRule", SimpleRule.class);
+        staticTypes.put("SimpleActionHandler", SimpleActionHandler.class);
+        staticTypes.put("SimpleConditionHandler", SimpleConditionHandler.class);
+
+        staticTypes.put("ActionHandlerFactory", ScriptedActionHandlerFactory.class);
+        staticTypes.put("ConditionHandlerFactory", ScriptedConditionHandlerFactory.class);
+        staticTypes.put("TriggerHandlerFactory", ScriptedTriggerHandlerFactory.class);
+
+        staticTypes.put("Configuration", Configuration.class);
+        staticTypes.put("Action", Action.class);
+        staticTypes.put("Condition", Condition.class);
+        staticTypes.put("Trigger", Trigger.class);
+        staticTypes.put("Rule", Rule.class);
+        staticTypes.put("ModuleType", ModuleType.class);
+        staticTypes.put("ActionType", ActionType.class);
+        staticTypes.put("TriggerType", TriggerType.class);
+        staticTypes.put("ConfigDescriptionParameter", ConfigDescriptionParameter.class);
+
+        types.addAll(staticTypes.keySet());
+
+        types.add("AutomationManager");
+        types.add("RuleRegistry");
+        types.add("rules");
+
+        presets.put("RuleSupport",
+                Arrays.asList("Configuration", "Action", "Condition", "Trigger", "Rule", "ModuleType", "ActionType"));
+        presets.put("RuleSimple", Arrays.asList("ScriptedRule", "SimpleRule"));
+        presets.put("RuleFactories", Arrays.asList("ActionHandlerFactory", "ConditionHandlerFactory",
+                "TriggerHandlerFactory", "TriggerType", "ConfigDescriptionParameter"));
+    }
+
+    public void setRuleRegistry(RuleRegistry ruleRegistry) {
+        this.ruleRegistry = ruleRegistry;
+    }
+
+    public void setRuleProvider(ScriptedRuleProvider ruleProvider) {
+        this.ruleProvider = ruleProvider;
+    }
+
+    protected void setItemRegistry(ItemRegistry itemRegistry) {
+        this.itemRegistry = itemRegistry;
+        staticTypes.put("ItemRegistry", this.itemRegistry);
```

Comment:
Are there languages that use object instances with a capital letter or why do you always add the same object with two different names? Can't we simply define a single name?